packages/core/admin/admin/src/components/ThemeToggleProvider.tsx

```diff
@@ -36,13 +32,30 @@ const ThemeToggleProvider = ({ children, themes }: ThemeToggleProviderProps) =>
     [setCurrentTheme]
   );
 
+  // Listen to changes in the system theme
+  React.useEffect(() => {
+    const themeWatcher = window.matchMedia('(prefers-color-scheme: dark)');
+    setSystemTheme(themeWatcher.matches ? 'dark' : 'light');
+
+    const listener = (event: MediaQueryListEvent) => {
+      setSystemTheme(event.matches ? 'dark' : 'light');
+    };
+    themeWatcher.addEventListener('change', listener);
+
+    // Remove listener on cleanup
+    return () => {
+      themeWatcher.removeEventListener('change', listener);
+    };
```

Comment:
I'm a bit confused by this part. Are we listening for a change in system theme? Is it going to change often? Wouldn't setting it once on mount be enough?

Reply:
Yes we're listening to changes to the system theme. I don't think the users change this tons of times, but when they do I think they expect everything to update at once. Otherwise it wouldn't live update like in my screen recording, which I think would be a bummer since it doesn't cost us much

Reply:
cool cool cool that works for me 🎈